### PR TITLE
[5.2] DB Query Builder Join Clause 'where in'

### PR DESCRIPTION
**Pull request by mightydes**
Added Join Clause methods:
- `whereIn`
- `whereNotIn`
- `orWhereIn`
- `orWhereNotIn`

[Issue discussion](https://github.com/laravel/framework/issues/4412)

### CODE REVIEW

File: src/Illuminate/Database/Query/JoinClause.php [hunk 1]
```diff
@@ -148,4 +148,52 @@ public function orWhereNotNull($column)
 		return $this->whereNotNull($column, 'or');
 	}
 
+	/**
+	 * Add an "on where in (...)" clause to the join.
+	 *
+	 * @param string $column
+	 * @param array $values
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/JoinClause.php]: You must have exactly 2 spaces either side of the type, not 1.

Comment (mightydes) [inline src/Illuminate/Database/Query/JoinClause.php]: @GrahamCampbell okay

**Comment by taylorotwell** [comment]
That nested ternary is real confusing. Also this should probaby go to 5.1 branch now.

**Comment by mightydes** [comment]
@taylorotwell
> That nested ternary is real confusing. Also this should probaby go to 5.1 branch now.

There you are, revise, please, [next](https://github.com/laravel/framework/pull/9016) "confuseless-if-else" version.

**Pull request closed without merging**